src/Runner.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -519,8 +518,18 @@ private async Task<bool> IsHostedServer(string serverUrl, VssCredentials credent
 
         private async Task<GitHubAuthResult> GetTenantCredential(string githubUrl, string githubToken, string runnerEvent)
         {
+            var githubApiUrl = "";
             var gitHubUrlBuilder = new UriBuilder(githubUrl);
-            var githubApiUrl = $"{gitHubUrlBuilder.Scheme}://api.{gitHubUrlBuilder.Host}/actions/runner-registration";
+            if (gitHubUrlBuilder.Host.Contains("github.com", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
have you considered string.equals instead?

Reply:
yes, i have, i guess i can do 
== github.com ||
== github.localhost ||
== www.github.com ||
== www.github.localhost